### PR TITLE
ref #126: add systemname field to message types

### PR DESCRIPTION
**Pull request by pixeljunkie**
| Q             | A
| ------------- | ---
| Branch        | 7.0.x
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#126
| License       | MIT

### CODE REVIEW

File: src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        'id' => '4',
+    ])
+;
+TCMSLogChange::update(__LINE__, $data);
+
+$query = "UPDATE `cms_message_manager_message_type` SET `systemname` = 'popup' WHERE `name` LIKE 'Popup%'";
+TCMSLogChange::RunQuery(__LINE__, $query);
+
+// generate additional systemnames from name field in lowercase and spaces replacement by underscores
+$query = "UPDATE `cms_message_manager_message_type` SET `systemname` = REPLACE(LOWER(`name`),' ','_') WHERE `systemname` = ''";
```
Comment (UlrichKu) [inline src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php:172]: Also this id seems to be fixed amongst systems.

File: src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+$data = TCMSLogChange::createMigrationQueryData('cms_message_manager_message_type', 'en')
+    ->setFields([
+        'systemname' => 'error',
+    ])
+    ->setWhereEquals([
+        'id' => '4',
+    ])
+;
+TCMSLogChange::update(__LINE__, $data);
```
Comment (UlrichKu) [inline src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php:166]: There is also the "error conspicuous".

**Comment by UlrichKu** [comment]
"Maybe" proposal: One could use the "name mapping table" approach from the other PR here still.

File: src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php (hunk 2 above)
Comment (pixeljunkie) [reply src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php:166]: I've added the error type with id 6 (ID 5 seems not to exist anywhere), but named it more common "error_striking"

File: src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php (hunk 1 above)
Comment (pixeljunkie) [reply src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php:182]: You mean popup? I have checked this and the id differs between projects, so i need to check for the name instead.

File: src/CoreBundle/private/library/classes/TCMSLogChange.class.php [hunk 3]
```diff
@@ -913,6 +915,16 @@ public static function GetMessageTypeByName($sMessageTypeName)
         return $sRet;
     }
 
+    public static function getMessageTypeIdBySystemName(string $messageTypeName): ?string
```
Comment (UlrichKu) [inline src/CoreBundle/private/library/classes/TCMSLogChange.class.php:918]: This method c/should document which system names there are normally.

File: src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+      'cms_field_type_id' => TCMSLogChange::GetFieldType('CMSFIELD_STRING'),
+      'isrequired' => '0',
+      'class_type' => 'Core',
+      'modifier' => 'none',
+      'restrict_to_groups' => '0',
+      'is_translatable' => '0',
+      'id' => 'f74894ff-4161-9f95-c390-0468ea94c338',
+  ])
+;
+TCMSLogChange::insert(__LINE__, $data);
```
Comment (UlrichKu) [inline src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php:22]: Set the position of that field?

Comment (pixeljunkie) [reply src/CoreBundle/Bridge/Chameleon/Migration/Script/update-1628503013.inc.php:22]: Is done in line 38.

File: src/CoreBundle/private/library/classes/TCMSLogChange.class.php (hunk 3 above)
Comment (pixeljunkie) [reply src/CoreBundle/private/library/classes/TCMSLogChange.class.php:918]: I've added a comment with the default available system names.

**Pull request merged**